SalesforceCommerceCloud/pwa-kit
# @W-15628148 Implement Pagination and Integrate Web Geolocation API

[W-15628148](https://gus.lightning.force.com/a07EE00001qFR1pYAG)
and
[W-15621326](https://gus.lightning.force.com/a07EE00001q8KlmYAE)

# Description
Added "Load More" button functionality to store locator and integrated native web geolocation API to "Use my Location" function. Made default of country selector, if defined in `constants.js`, to show "Select a country". Updated UI text to remove distance and show information on geolocation. 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1) Go to https://runtime-admin-n16.mobify-storefront.com/ to check out the deployment
- (step2) Open the store locator modal by pressing the icon on the top right of the home page
- (step3) See that results from the default geolocation set by the merchant is shown. The input fields have an empty postal code and a "Select a country" for the country selector. (NOTE: you will see results from different countries because for testing purposes I have set the distance to look for stores around a given location to be the API maximum)
### Load More testing
- (step4) Scroll down and press load more to see that 10 more stores are loaded
- (step5) Keep pressing load more until there are no more stores in the list
### Use my location testing
- (step6) Press use my location to find the permission dropdown from your browser show up. Allow sharing.
- (step7) See that UI text changes to "Viewing stores near your location", and shows the store closest to you first.
- (step8) Close and open the modal to see that the modal keeps your last input (i.e. before close if you had pressed "Use My Location" it will show you location around you. If you had given a manual input, e.g. Germany 24103, then it will show you location around there. 
### Testing the store locator page
- (step9) Go to https://runtime-admin-n16.mobify-storefront.com/global/en-GB/store-locator 
- (step10) Now that you have your location sharing enabled, see that it shows you stores around you. Test the Load More functionality like in step4-step5
- (step11) The behavior of the store locator page and the modal are different. Closing the store locator page and opening does not result in the remembering of previous input. Check that this is the case.
- (step12) Alternate between providing manual input and pressing "Use my Location" to test the functionalities.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
